File: app/src/main/java/tomaszkopacz/meetbam/activities/AccountPhotosFragment.java
package tomaszkopacz.meetbam.activities;


import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.StaggeredGridLayoutManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import tomaszkopacz.meetbam.R;
import tomaszkopacz.meetbam.model.Post;
import tomaszkopacz.meetbam.presenters.AccountPhotosFragmentPresenter;
import tomaszkopacz.meetbam.service.PostAdapter;
import tomaszkopacz.meetbam.service.WebService;

/**
 * A simple {@link Fragment} subclass.
 */
public class AccountPhotosFragment extends Fragment {

    private AccountPhotosFragmentPresenter presenter;

    @Inject
    WebService service;

    @BindView(R.id.userPostsRecView)
    RecyclerView recView;

    private PostAdapter adapter;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        //inflate the layout for this fragment
        View view
                = inflater.inflate(R.layout.fragment_account_photos, container, false);

        //bind views
        ButterKnife.bind(this, view);

        //get dependencies
        ((MainApp)(getActivity().getApplication()))
                .getWebServiceComponent().inject(this);

        //set up presenter
        presenter = new AccountPhotosFragmentPresenter(this, service);

        //prepare recycler view and download posts
        prepareRecView();
        presenter.downloadPostsList();

        return view;
    }

    /**
     * Configures recycler view. Recycler has fixed size, two spans oriented vertically
     * and default item animator.
     */
    private void prepareRecView(){
        recView.setHasFixedSize(true);
        recView.setLayoutManager(new StaggeredGridLayoutManager(2,
                StaggeredGridLayoutManager.VERTICAL));
        recView.setItemAnimator(new DefaultItemAnimator());
    }

    /**
     * Sets list of posts to recycler view.
     */
    public void setUpList(List<Post> posts) {
        if (posts == null)
            posts = new ArrayList<>();

        adapter = new PostAdapter(this.getActivity(), posts, recView);
        recView.setAdapter(adapter);
    }
}

File: app/src/main/java/tomaszkopacz/meetbam/view/AccountPhotosFragment.kt
package tomaszkopacz.meetbam.view


import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.StaggeredGridLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_account_photos.*
import tomaszkopacz.meetbam.R
import tomaszkopacz.meetbam.presenter.AccountPhotosFragmentPresenter

class AccountPhotosFragment : Fragment() {

    private lateinit var presenter: AccountPhotosFragmentPresenter

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_account_photos, container, false)

        presenter = AccountPhotosFragmentPresenter(this)

        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        prepareRecView()
        presenter.downloadPostsList()
    }

    private fun prepareRecView() {
        user_posts_recview.layoutManager = StaggeredGridLayoutManager(2,
                StaggeredGridLayoutManager.VERTICAL)
        user_posts_recview.setHasFixedSize(true)
        user_posts_recview.itemAnimator = DefaultItemAnimator()
    }

    fun putPhotos(adapter: PostAdapter) {
        user_posts_recview.adapter = adapter
    }
}
